### DISCUSSION TITLE
ScenegraphLayer. Animation management

### DISCUSSION THREAD

**Discussion by VladMonk** [visgl/deck.gl#8165] [Q&A]
### Hello everyone! 

I'm working on a demo now.  There is a simple 3d scene with different  **.glb** objects displayed on top of the maplibre basemap. At this momemt, when i need to switch animation, i'm delete layer and create new one with new object, wich actualy the same as previous, but with different animation. Thats cringe little bit i know)

So, what i want? 

- There is a way to handle animation cycles? I mean check when animation end and, for example, stop the loop
- Get list of animations names, to place them instead of **"*"**
```
    _animations: {
      '*': {speed: 1}
    },
```
Also i've tried names from blender, it doesn't work
![изображение](https://github.com/visgl/deck.gl/assets/56898512/feb81177-a2b6-4d92-9281-abff8bb89000)

-  ![изображение](https://github.com/visgl/deck.gl/assets/56898512/0f44d633-3a3f-446b-8e1b-d462b8333898) find this note in changelog. Interested in the sorting animation order. I guess this is related to the one above

P.S. Maybe should refuse animation in blender and do it in JS? There is option like this?

**Comment by ibgreen** [comment]
Agree that more work could be done on exposing glTF animations to the application. The support (e.g. in terms of additional callbacks/query functions) should probably go into luma.gl, perhaps with some minimal glue code in deck.gl.

The glTF support in luma.gl is relatively old and is getting ripe for a refactor in v9. Meaning that adding it in v8 could be a bit messy and not translate directly to v9. 

Are you in a mood to contribute? If not we can open these requirements on luma.gl and see if something happens.

**Reply by VladMonk** [reply]
Thanks! I'll keep in mind about the luma versions. I'm not sure if my participation will usefull, sorry)
